src/components/AddItem.js

```diff
@@ -21,30 +42,32 @@ const AddItem = () => {
     e.preventDefault();
 
     shoppingLists
-      // fetch the token from localSorage
       .where('token', '==', getToken())
       .get()
       .then((data) => {
-        // if the shoppingList with the token 801 exists
         if (data.docs.length) {
-          shoppingLists
-            .doc(data.docs[0].id)
-            // just append the new item to that shoppingList items
-            .update({
-              items: firebase.firestore.FieldValue.arrayUnion(newItem),
-            });
+          const itemsNames = getItemsNamesFromDoc(data.docs[0]);
+
+          if (existingName(itemsNames, newItem.name)) {
+            displayMessage(
+              `The item: ${newItem.name} already exists!!`,
+              'error',
+            );
+            return;
+          }
+
+          shoppingLists.doc(data.docs[0].id).update({
+            items: firebase.firestore.FieldValue.arrayUnion(newItem),
+          });
         } else {
-          // else just create a new list and add that item to it
           shoppingLists.add({
-            // fetch the token from localSorage
             token: getToken(),
             items: [newItem],
           });
         }
-        alert('Successfully Added');
-        setInputValue('');
-        // reset radio buttons
-        document.getElementById('soon').checked = true;
+
+        displayMessage('Successfully Added', 'success');
+        resetForm();
```

Comment:
This is so cool! Passing the message and the type of it to `displayMessage` and declaring `resetForm()` for resetting the form. 👍